src/renderer/base/renderer.cpp

```diff
@@ -466,17 +466,22 @@ void Renderer::TriggerScroll(const COORD* const pcoordDelta)
 // - <none>
 void Renderer::TriggerCircling()
 {
+    const auto rects = _GetSelectionRects();
+
     for (IRenderEngine* const pEngine : _rgpEngines)
     {
         bool fEngineRequestsRepaint = false;
         HRESULT hr = pEngine->InvalidateCircling(&fEngineRequestsRepaint);
         LOG_IF_FAILED(hr);
 
+        LOG_IF_FAILED(pEngine->InvalidateSelection(rects));
+
         if (SUCCEEDED(hr) && fEngineRequestsRepaint)
         {
             LOG_IF_FAILED(_PaintFrameForEngine(pEngine));
         }
     }
+    _pData->CircleSelections();

```

Comment:
I'm a bit concerned about this, actually. I think this is the **first** instance where the renderer can _change data in the terminal_. I don't think we should ever allow that; rather, we should handle this before it even gets to the renderer.

(Aside: the renderer telling us about circling so that we can do something about it is a very long game of telephone. Terminal -> Buffer -> Renderer -> Terminal again.)

Reply:
*facepalm* yes that makes much more sense

Reply:
fixed